api/src/main/java/io/minio/credentials/IamAwsProvider.java

```diff
@@ -140,6 +140,7 @@ private String fetchImdsToken() {
           new HttpUrl.Builder()
               .scheme(url.scheme())
               .host(url.host())
+              .port(url.port())
```

Comment:
Add port only if port and service do not match i.e. ignore port when port and service matches i.e HTTP -> 80, HTTPS -> 443

Reply:
Is there any argument to do it this way? To my best knowledge applying the default port doesn't affect the call in any negative way. Any additional logic makes this feature more complicated to understand and maintain for future developers. 

Ofc if you insist, i'll change it, but i'd love to see a rationale behind this

Reply:
It is required for Signature v4 computation.

Reply:
Are IAM requests signed? I cannot recall seeing any signing in IamAwsProvider and cannot recall this in AWS docs. To my best knowledge these are plain HTTP requests

Reply:
For `IamAwsProvider`, the request is not signed. However it is better to add port only if it is not standard port for correctness.

Reply:
As you wish. Uploaded new commit with the fix